modules/spring-config/src/main/java/org/mule/config/spring/CompositeOptionalObjectsController.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.config.spring;
+
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.util.ArrayUtils;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Implementation of {@link OptionalObjectsController} which groups
+ * a list of controllers and executes every operation on all of them.
+ *
+ * @since 3.7.0
+ */
+public class CompositeOptionalObjectsController implements OptionalObjectsController
+{
+
+    private final List<OptionalObjectsController> controllers;
+
+    public CompositeOptionalObjectsController(OptionalObjectsController... controllers)
+    {
+        checkArgument(!ArrayUtils.isEmpty(controllers), "cannot compose empty controllers list");
+        this.controllers = ImmutableList.copyOf(controllers);
+    }
+
+    @Override
+    public void registerOptionalKey(String key)
+    {
+        for (OptionalObjectsController controller : controllers)
+        {
+            controller.registerOptionalKey(key);
+        }
+    }
+
+    @Override
+    public void discardOptionalObject(String key)
+    {
+        for (OptionalObjectsController controller : controllers)
+        {
+            controller.discardOptionalObject(key);
+        }
+    }
+
+    @Override
+    public boolean isOptional(String key)
+    {
+        for (OptionalObjectsController controller : controllers)
```

Comment:
no, it has to be on at least one. If the optional entry is in the domain, then the one from the app will return false but you still have to ignore the exception